File: src/main/java/com/gitlab/daring/image/command/structure/DetectLinesCommand.java
package com.gitlab.daring.image.command.structure;

import com.gitlab.daring.image.command.BaseCommand;
import com.gitlab.daring.image.command.CommandEnv;
import org.bytedeco.javacpp.opencv_core.Mat;
import org.bytedeco.javacpp.opencv_imgproc.LineSegmentDetector;

import static com.gitlab.daring.image.util.ImageUtils.newScalarMat;
import static org.bytedeco.javacpp.opencv_imgproc.createLineSegmentDetector;

public class DetectLinesCommand extends BaseCommand {

	//TODO support all params
	final LineSegmentDetector detector = createDetector();

	final Mat lines = new Mat();

	public DetectLinesCommand(String... args) {
		super(args);
	}

	@Override
	public void execute(CommandEnv env) {
		detector.detect(env.mat, lines);
		env.mat.setTo(newScalarMat(0));
		detector.drawSegments(env.mat, lines);
	}

	private LineSegmentDetector createDetector() {
		return createLineSegmentDetector();
	}

}

File: src/main/kotlin/com/gitlab/daring/image/command/structure/DetectLinesCommand.kt
package com.gitlab.daring.image.command.structure

import com.gitlab.daring.image.command.CommandEnv
import com.gitlab.daring.image.command.KBaseCommand
import com.gitlab.daring.image.opencv.setTo
import org.bytedeco.javacpp.opencv_core.Mat
import org.bytedeco.javacpp.opencv_imgproc.createLineSegmentDetector

internal class DetectLinesCommand(args: Array<String>) : KBaseCommand(args) {

    //TODO support all params
    val detector = createLineSegmentDetector()

    val lines = Mat()

    override fun execute(env: CommandEnv) {
        detector.detect(env.mat, lines)
        env.mat.setTo(0)
        detector.drawSegments(env.mat, lines)
    }

}